### PR TITLE
(chore) Security Updates

### PR DESCRIPTION
**Pull request by strobelm**
Hey all,

I've prepared a pull request with some security updates for the meldeplattform. Here's a breakdown of the changes:

1. **Ran `npm audit fix`**: Ensure that we fix any vulnerabilities reported by npm audit.

2. **Deduplicate version strings in Dockerfile and update to latest versions of Go and Alpine**: Optimize our Dockerfile by removing duplicate version strings and ensuring we're using the latest versions of Go and Alpine.

3. **Update all Go dependencies recursively**: Ensure we're using the latest security patches and bug fixes by updating all dependencies recursively.

**Impact**

Closes several security vulnerabilities in the go and node packages. No functional changes were done.

**How to test**

I was running this using `podman run --privileged -v ./files:/files -p 8080:8080 `, the page seems to load fine. One thing I noticed using this local setup is that I actually cannot really select an issue. Any input on this?

image

### CODE REVIEW

**Comment by joschahenningsen** [comment]
> the page seems to load fine. One thing I noticed using this local setup is that I actually cannot really select an issue. Any input on this?

 Can you verify that the config.yaml is mounted next to the executable within the container?  Any reason you use podman over Docker, considering Docker is well maintained and documented for this application?

File: go.mod [hunk 1]
```diff
@@ -1,52 +1,63 @@
 module github.com/TUM-Dev/meldeplattform
 
-go 1.19
+go 1.21
```
Comment (joschahenningsen) [inline go.mod:3]: Please use go 1.22

File: Dockerfile [hunk 2]
```diff
@@ -1,10 +1,13 @@
-FROM node:current-alpine3.17 as web
+ARG ALPINEVERSION="3.18"
+ARG GOVERSION="1.22"
```
Comment (joschahenningsen) [inline Dockerfile:2]: Don't see a real reason for this but works for me

**Review by joschahenningsen — CHANGES_REQUESTED** [review]

**Comment by joschahenningsen** [comment]
> One thing I noticed using this local setup is that I actually cannot really select an issue. Any input on this?

When running this locally, the database will initially be empty and therefore there are no topics to report to. In order to create some you have to log in and create one. This could be a little tricky tho because you don't have the idp certificate locally... Do you want me to build a simple debug mode that allows local admin login in development environments for testing?

**Review by joschahenningsen — APPROVED** [review]

**Comment by strobelm** [comment]
> > One thing I noticed using this local setup is that I actually cannot really select an issue. Any input on this?
> 
> When running this locally, the database will initially be empty and therefore there are no topics to report to. In order to create some you have to log in and create one. This could be a little tricky tho because you don't have the idp certificate locally... Do you want me to build a simple debug mode that allows local admin login in development environments for testing?

Thanks for the offer, but I think I can manage :slightly_smiling_face:. One more thing I could use your help: I do not have the rights to push the merge button :smile:

**Pull request merged**